doc/monitoring_containers/monitoring_containers.md

```diff
@@ -0,0 +1,276 @@
+* [2 Functionality](#2-functionality)
+    - [2.1 Target Deployment Use Cases](#21-target-deployment-use-cases)
+    - [2.2 Functional Description](#22-functional-description)
+        - [2.2.1 Monitoring Critical Processes](#221-monitoring-critical-processes)
+        - [2.2.2 Monitoring Critical Resource Usage](#222-monitoring-critical-resource-usage)
+        - [2.2.3 Auto-restart Docker Container](#223-auto-restart-docker-container)
+            - [2.2.3.1 CLI (and usage example)](#2231-cli-and-usage-example)
+                - [2.2.3.1.1 Show the Status of Auto-restart](#22311-show-the-status-of-auto-restart)
+                - [2.2.3.1.2 Configure the Status of Auto-restart](#22312-configure-the-status-of-auto-restart)
+                - [2.2.3.1.3 CONTAINER_FEATURE Table](#22313-container_feature-table)
+
+# List of Tables
+* [Table 1: Abbreviations](#definitionsabbreviation)
+
+# Revision
+| Rev |    Date    |          Author        |     Change Description    |
+|:---:|:----------:|:----------------------:|---------------------------|
+| 0.1 | 02/18/2020 | Yong Zhao, Joe Leveque |      Initial version      |
+
+# About this Manual
+This document presents the design and implementation of feature to monitor and auto-mitigate
+the unhealthy of docker containers in SONiC.
+
+# Scope
+This document describes the high level design of feature to monitor and auto-mitigate
+the unhealthy of docker containers.
+
+# Definitions/Abbreviation
+| Abbreviation |         Description          |
+|--------------|------------------------------|
+| Config DB    | SONiC Configuration Database |
+| CLI          | Command Line Interface       |
+
+# 1 Feature Overview
+SONiC is a collection of various switch applications which are held in docker containers
+such as BGP container and SNMP container. Each application usually includes several processes which are 
+working together to provide and receive the services from other modules. As such, the healthy of
+critical processes in each docker container are the key not only for the docker
+container working correctly but also for the intended functionalities of entire SONiC switch.
+On the other hand, profiling the resource usages and performance of each docker
+container are also important for us to understand whether this container is in healthy state
+or not and furtherly to provide us with deep insight about networking traffic.
+
+The main purpose of this feature includes two parts: the first part is to monitor the
+running status of each process and critical resource usage such as CPU, memory and disk
+of each docker container.
+The second part is docker containers can be automatically shut down and
+restarted if one of critical processes running in the container exits unexpectedly. Restarting
+the entire container ensures that configuration is reloaded and all processes in the container
+get restarted, thus increasing the likelihood of entering a healthy state.
```

Comment:
I think these "parts" are not very clear. I think you need to make a clearer separation between the monitoring and auto-mitigation features.

Reply:
Very good suggestion! I split this part into two sections which describe monitoring and auto-mitigating respectively.